### DISCUSSION TITLE
Atari: Move ONCE into separate chunk

### DISCUSSION THREAD

**Discussion by polluks2** [cc65/cc65#1525] [Ideas]
Is it useful and possible to move the ONCE segment into a separate chunk? Like the XL check.

**Comment by groessler** [comment]
Hmm. I had thought about that before as well.

I think some things which are in ONCE could be moved to a separate chunk (determine DOS version). With some other things I have doubts (IRQ setup).

How is it done on the other targets? How do they reclaim the "ONCE" memory?

**Comment by polluks2** [comment]
I suppose other ABI's are not smart enough, so it would be of no benefit.

**Comment by oliverschmidt** [comment]
> How is it done on the other targets? How do they reclaim the "ONCE" memory?

The "usual" approach is to re-use the `ONCE` memory for the `BSS`. See e.g. https://github.com/cc65/cc65/blob/master/cfg/apple2.cfg.

**Reply by polluks2** [reply]
I see, however this chunk could become more valuable `CODE`.

**Reply by groessler** [reply]
I like the idea, but I'm worried that this would complicate the Atari XL linker configs even more.
